cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+        cl_cleaned_opinions.append(
+            {
+                "id": op.id,
+                "byline": op.author_str,
+                "type": op.type,
+                "opinion": prep_text,
+            }
+        )
+
+    return cl_cleaned_opinions
+
+
+def update_matching_opinions(
+    matches: dict, cl_cleaned_opinions: list, columbia_opinions: list
+) -> None:
+    """Store matching opinion content in html_columbia
+    :param matches: dict with matching position from cl and columbia opinions
+    :param cl_cleaned_opinions: list of cl opinions
+    :param columbia_opinions: list of columbia opinions
+    """
+    for columbia_pos, cl_pos in matches.items():
+        file_opinion = columbia_opinions[columbia_pos]  # type: dict
+        file_byline = file_opinion.get("byline")
+        cl_opinion = cl_cleaned_opinions[cl_pos]
+        opinion_id_to_update = cl_opinion.get("id")
+
+        op = Opinion.objects.get(id=opinion_id_to_update)
+        author_str = ""
+
+        if file_byline:
+            # Prettify the name a bit
+            author_str = titlecase(find_just_name(file_byline.strip(":")))
+
+        if op.author_str == "":
+            # We have an empty author name
+            if author_str:
+                # Store the name extracted from the author tag
+                op.author_str = author_str
+        else:
+            if author_str:
+                if (
+                    find_just_name(op.author_str).lower()
+                    != find_just_name(author_str).lower()
+                ):
+                    # last resort, use distance between words to solve typos
+                    s = SequenceMatcher(
+                        None,
+                        find_just_name(op.author_str).lower(),
+                        find_just_name(author_str).lower(),
+                    )
+                    if s.ratio() >= 0.6:
+                        # columbia names are better
+                        op.author_str = author_str
+                    else:
+                        raise AuthorException(f"Authors don't match")
+                elif any(s.isupper() for s in op.author_str.split(",")):
+                    # Some names are uppercase, update with processed names
+                    op.author_str = author_str
+
+        converted_text = convert_columbia_html(
+            file_opinion["opinion"], columbia_pos
+        )
+        op.html_columbia = str(converted_text)
+        op.save()
+
+
+def match_text_lists(
+    file_opinions_list: list[Any], cl_opinions_list: list[Any]
+) -> dict[int, int]:
+    """Generate matching lists above threshold
+    :param file_opinions_list: Opinions from file
+    :param cl_opinions_list: CL opinions
+    :return: Matches if found or empty dict
+    """
+    # We import this here to avoid a circular import
+    from cl.corpus_importer.management.commands.harvard_opinions import (
+        compare_documents,
```

Comment:
I think we need a better name here.

Reply:
I tried a new name, I think this is the most difficult 🙃